olive/engine/engine.py

```diff
@@ -810,15 +816,19 @@ def _run_pass(
             accelerator_spec,
         ]
         output_model_id = "-".join(map(str, output_model_id_parts))
-        output_model_path = str(self._model_cache_path / f"{output_model_id}")
+        output_model_path = self._model_cache_path / f"{output_model_id}" / "output_model"
+        output_model_path.parent.mkdir(parents=True, exist_ok=True)
+        output_model_path = str(output_model_path)
 
         # prune if invalid search_point
         if not p.validate_search_point(pass_search_point) and not self.no_search:
             output_model = PRUNED_CONFIG
         else:
             # run pass
+            host = self.host_for_pass(pass_id)
+            if host.system_type != SystemType.AzureML:
+                input_model = self._prepare_non_local_model(input_model)
```

Comment:
need the input_model_id be updated?

Reply:
The input model_id is the same. What this method does is to download the remote resource (or load from cache) and update the model object's `local_resource_path` attribute. this attribute is used to get `model.model_path`